server/repositories/mongo/mongo-messages-repository.ts

```diff
@@ -0,0 +1,392 @@
+    .toArray()
+    .then((result: Message[]) => {
+      console.log(result.length);
+      return result;
+    });
+};
+
+const getMessagesByAction: (
+  space: string,
+  actionId: string,
+  msgType: string
+) => Promise<Message[]> = async (space, actionId, msgType) => {
+  const db = await snapshotHubMongoDb;
+  const messagesCollection = db.collection(
+    process.env.MONGODB_TRIBUTE_DAOS_SNAPSHOT_HUB_DB_MESSAGES_COLLECTION_NAME
+  );
+  return messagesCollection
+    .find<Message>(
+      { space, type: msgType, actionId },
+      { sort: { timestamp: -1 } }
+    )
+    .toArray()
+    .then((result: Message[]) => {
+      console.log(result.length);
+      return result;
+    });
+};
+
+const getMessagesById: (
+  space: string,
+  id: string,
+  msgType: string
+) => Promise<Message[]> = async (space, id, msgType) => {
+  const db = await snapshotHubMongoDb;
+  const messagesCollection = db.collection(
+    process.env.MONGODB_TRIBUTE_DAOS_SNAPSHOT_HUB_DB_MESSAGES_COLLECTION_NAME
+  );
+  return messagesCollection
+    .find<Message>({ space, type: msgType, id })
+    .toArray()
+    .then((result: Message[]) => {
+      console.log(result.length);
+      return result;
+    });
+};
+
+const getVoteBySender: (
+  space: string,
+  address: string,
+  proposalId: string
+) => Promise<Message[]> = async (space, address, proposalId) => {
+  const db = await snapshotHubMongoDb;
+  const messagesCollection = db.collection(
+    process.env.MONGODB_TRIBUTE_DAOS_SNAPSHOT_HUB_DB_MESSAGES_COLLECTION_NAME
+  );
+  return messagesCollection
+    .find<Message>({
+      space,
+      address,
+      'payload.proposalId': proposalId,
+      type: 'vote'
+    })
+    .toArray()
+    .then((result: Message[]) => {
+      console.log(result.length);
+      return result;
+    });
+};
+
+const getProposalByDraft: (
+  space: string,
+  id: string
+) => Promise<Message[]> = async (space, id) => {
+  const db = await snapshotHubMongoDb;
+  const messagesCollection = db.collection(
+    process.env.MONGODB_TRIBUTE_DAOS_SNAPSHOT_HUB_DB_MESSAGES_COLLECTION_NAME
+  );
+  return messagesCollection
+    .find<Message>({
+      space,
+      'data.erc712DraftHash': id,
+      type: 'proposal'
+    })
+    .toArray()
+    .then((result: Message[]) => {
+      console.log(result.length);
+      return result;
+    });
+};
+
+const getProposalVotes: (
+  space: string,
+  id: string
+) => Promise<Message[]> = async (space, id) => {
+  const db = await snapshotHubMongoDb;
+  const messagesCollection = db.collection(
+    process.env.MONGODB_TRIBUTE_DAOS_SNAPSHOT_HUB_DB_MESSAGES_COLLECTION_NAME
+  );
+  return messagesCollection
+    .find<Message>(
+      {
+        space,
+        'payload.proposalId': id,
+        type: 'vote'
+      },
+      { sort: { timestamp: 1 } }
+    )
+    .toArray()
+    .then((result: Message[]) => {
+      console.log(result.length);
+      return result;
+    });
+};
+
+const getAllProposalsAndVotes: (
```

Comment:
For this and `getAllProposalsAndVotesByAction ` do you wish to use `findVotesForProposals` in the implementation, like in the Postgres adapter versions, or are you planning on using it elsewhere?

Later, we can write an aggregation query maybe rather than running a `map` - oops. Could've done something similar in Postgres (ah, hindsight).